### PR TITLE
feat: e2e test server initial implementation

### PR DESCRIPTION
**Pull request by aabmass**
This implements the e2e test server we already have in Java and Python ops repos. No scenarios are implemented here, just the boilerplate.

### CODE REVIEW

File: e2e-test-server/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Build relative to root of repository i.e. `docker build --file Dockerfile --tag=$tag ..`
+
+FROM node:15-alpine as node-base
```
Comment (weyert) [inline e2e-test-server/Dockerfile:17]: Isn't v15 already deprecated?

Comment (aabmass) [reply e2e-test-server/Dockerfile]: Yup. I'll make a PR to update all of our CI with the latest versions. Thanks!

File: e2e-test-server/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      [constants.STATUS_CODE]: statusCode.toString(),
+    });
+  }
+
+  async function onRequestMessage(message: Message): Promise<void> {
+    const testId = message.attributes?.[constants.TEST_ID];
+    if (testId === undefined) {
+      // don't even know how to write back to the publisher that the message is
+      // invalid, so nack()
+      message.nack();
```
Comment (punya) [inline e2e-test-server/src/index.ts:49]: Should we return early after doing this?

File: e2e-test-server/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    const scenario = message.attributes?.[constants.SCENARIO];
+    if (scenario === undefined) {
+      respond(testId, {
+        statusCode: Status.INVALID_ARGUMENT,
+        data: Buffer.from(
+          `Expected attribute ${constants.SCENARIO} is missing`
+        ),
+      });
+    }
```
Comment (punya) [inline e2e-test-server/src/index.ts:60]: Same question - return from here?

File: e2e-test-server/src/scenarios.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  statusCode: Status;
+  data?: Buffer;
+}
+
+async function health(): Promise<Response> {
+  return {statusCode: Status.OK};
+}
+
+async function notImplementedHandler(): Promise<Response> {
+  return {statusCode: Status.UNIMPLEMENTED};
```
Comment (punya) [inline e2e-test-server/src/scenarios.ts:33]: Is this supposed to be `Promise.resolve(...)`?

File: e2e-test-server/src/index.ts (hunk 2 above)
Comment (aabmass) [reply e2e-test-server/src/index.ts:49]: yup thank you 🤦

File: e2e-test-server/src/scenarios.ts (hunk 4 above)
Comment (aabmass) [reply e2e-test-server/src/scenarios.ts:33]: The `async` makes the return type a promise, so it's not needed

**Review by punya — APPROVED** [review]

**Pull request merged**